docs/website/docs/development/architecture/how-odo-works.md

```diff
@@ -0,0 +1,1382 @@
+---
+title: How odo works
+sidebar_position: 30
+# Display h2 to h5 headings
+toc_min_heading_level: 2
+toc_max_heading_level: 6
+---
+
+## How `odo dev` works
+
+In a nutshell, when running [`odo dev`](../../command-reference/dev), `odo` will do the following:
+
+1. It reads and validates the Devfile in the current directory.
+   For example, it makes sure a `command` of the right kind (`run` or `debug` if running `odo dev --debug`) is defined.
+2. It determines the resources it needs to create (or update) and manage, and instructs the Kubernetes API Server to create all necessary resources.
+   Specifically, `odo` creates the following resources:
+   - [Deployment](https://kubernetes.io/docs/concepts/workloads/controllers/deployment/) for running the containers. See [below](#deployment) for further details.
+   - [Service](https://kubernetes.io/docs/concepts/services-networking/service/) for accessibility. See [below](#service) for further details.
+
+   If the Devfile contains any [`Kubernetes` components](../../development/devfile#components) not referenced by any [`apply` command](https://devfile.io/docs/2.2.0-alpha/adding-apply-commands),
+   `odo dev` automatically applies them.
+   This is the mechanism used by [`odo add binding`](../../command-reference/add-binding) to have `ServiceBinding` resources created by `odo dev`. 
```

Comment:
I was hesitant to add a dedicated section about how `odo` handles Devfile `kubernetes` components. But you're right it is currently a bit confusing. Removing this sentence for now - this could be the scope of a separate PR.